lib/rubocop/cop/mixin/surrounding_space.rb

```diff
@@ -98,40 +88,29 @@ def space_offenses(node, # rubocop:disable Metrics/ParameterLists
         unless extra_space?(left_token, :left) || start_ok
           space_offense(node, left_token, :none, message, SPACE_COMMAND)
         end
-        return if right_bracket_ok?(right_token) || end_ok
+        return if extra_space?(right_token, :right) || end_ok
         space_offense(node, right_token, :none, message, SPACE_COMMAND)
       end
 
       def space_corrector(corrector, left_token, right_token)
-        unless space_after?(left_token)
+        unless left_token.space_after?
           corrector.insert_after(left_token.pos, ' ')
         end
-        return if space_before?(right_token)
+        return if right_token.space_before?
         corrector.insert_before(right_token.pos, ' ')
       end
 
       private
 
       def extra_space?(token, side)
+        return false unless token
         if side == :left
-          extra_space_after?(token)
+          String(token.space_after?) == ' '
         else
-          extra_space_before?(token)
+          String(token.space_before?) == ' '
         end
       end
 
-      def extra_space_after?(token)
-        token && String(space_after?(token)) == ' '
-      end
-
-      def extra_space_before?(token)
-        token && String(space_before?(token)) == ' '
-      end
-
-      def right_bracket_ok?(token)
-        extra_space?(token, :right) || token.nil?
-      end
-
```

Comment:
The removal of these 3 methods is only tangential to moving the space_before/after methods. This related cleanup was made possible by the employing a better right_bracket guard clause discussed [here](https://github.com/bbatsov/rubocop/pull/5149#discussion_r155824009).